components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/username/UsernameRecoveryManagerImpl.java

```diff
@@ -355,11 +356,23 @@ private void triggerNotification(User user, String notificationChannel, String e
 
         String combinedUsernames = user.getUserName();
         String[] usernames = combinedUsernames.split(",");
+        String[] userStoreDomains = null;
+        if (user.getUserStoreDomain() != null) {
+            userStoreDomains = user.getUserStoreDomain().split(",");
+        }
+
+        int userIndex = 0;
         for (String username : usernames) {
             HashMap<String, Object> properties = new HashMap<>();
             properties.put(IdentityEventConstants.EventProperty.USER_NAME, username);
             properties.put(IdentityEventConstants.EventProperty.TENANT_DOMAIN, user.getTenantDomain());
-            properties.put(IdentityEventConstants.EventProperty.USER_STORE_DOMAIN, user.getUserStoreDomain());
+
+            String userStoreDomain = user.getUserStoreDomain();
+            if (userStoreDomains != null && userStoreDomains.length > userIndex) {
+                userStoreDomain = userStoreDomains[userIndex];
```

Comment:
The logic assumes usernames[userIndex] and userStoreDomains[userIndex] are always aligned right? If so is the PRIMARY user store domain name also included in the userStoreDomains array?

Reply:
Yes it has to assume that information is retrieved in order.
The user object is formulated as 
username: john, smith
userStoreDomain: primary, secondary